platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/collections/LongListOffHeapTest.java

```diff
@@ -117,4 +118,30 @@ public void testPersistListWithNonZeroMinValidIndex(@TempDir final Path tempDir)
             }
         }
     }
+
+    @Test
+    public void testFileFormatBackwardCompatibility_halfEmpty() throws URISyntaxException, IOException {
+        final Path pathToList = ResourceLoader.getFile("test_data/LongListOffHeapHalfEmpty_10k_10pc_v1.ll");
```

Comment:
Rather than store binary files in resources for tests, it _might_ be better to have a small class that creates the file (as a temp file) to be loaded...

Reply:
In general, I agree - it's better to have generated data instead of canned data but in this particular case I'm testing backward compatibility and incorrect versions. To generate a file of the old version I'd have to keep the old version of the class in the codebase. I don't think it's worth it.

Reply:
To an extent, perhaps.  I would probably have a class in tests that generates the test files, mostly independently, and that would always be able to generate the current version and one older version.  Its only purpose being to support compatible migration.
Not required, by any means, but might be useful and a bit more likely to be properly updated in any future versioned updates.